lib/rubocop/formatter/offense_count_formatter.rb

```diff
@@ -37,6 +38,9 @@ def started(target_files)
 
       def file_finished(_file, offenses)
         offenses.each { |o| @offense_counts[o.cop_name] += 1 }
+        if options[:display_style_guide]
+          offenses.each { |o| @style_guide_links[o.cop_name] ||= o.message[/ \(http\S+\)\Z/] }
```

Comment:
I went for a simple solution here, parsing the message rather than adding a style guide link property to the Offense class.